WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -17,8 +17,11 @@ class CompetitionsController < ApplicationController
     :admin_edit,
   ]
 
-  private def competition_from_params
-    Competition.find(params[:competition_id] || params[:id]).tap do |competition|
+  private def competition_from_params(include_events: false, include_rounds: false)
```

Comment:
Instead of using parameters for various associations to include, why not let the caller pass the list of associations to include?
I did something similar [here](https://github.com/viroulep/wc2017/blob/master/app/models/registration.rb#L67), it could be more flexible if at some point we need to include other associations.